source/dotnet/Library/AdaptiveCards/AdaptiveBaseElement.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Xml.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveInternalID
+    {
+        public static uint Invalid = 0;
+        private static uint CurrentInternalID = 1;
+        private uint ID { get; }
+
+        public AdaptiveInternalID()
+        {
+            ID = Invalid;
+        }
+
+        private AdaptiveInternalID(uint id)
+        {
+            ID = id;
+        }
+
+        public static AdaptiveInternalID Current()
+        {
+            return new AdaptiveInternalID(CurrentInternalID);
+        }
+
+        public static AdaptiveInternalID Next()
```

Comment:
is this class threadsafe? Does this need to be ?

Reply:
no on both counts. we provide no thread safety guarantees currently, and have a non-trivial amount of work to do if we end up needing it. :)

---
In reply to: [278316503](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278316503) [](ancestors = 278316503)